lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,20 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      paths = Set.new(applicable_files.map { |file| File.dirname(file) })
```

Comment:
`File.dirname` returns the directory without a trailing slash. `git ls-tree` only returns all files in the directory if the directory name has a trailing `/` (I'm running git 1.8.3.1). Did you test this?